data_safe_haven/commands/shm.py

```diff
@@ -142,6 +142,14 @@ def teardown() -> None:
     try:
         config = SHMConfig.from_remote(context)
         shm_infra = ImperativeSHM(context, config)
+        console.print(
+            "Tearing down the Safe Haven Management environment will permanently delete all associated resources, including remotely stored configurations."
+        )
+        if not console.confirm(
+            "Do you wish to continue tearing down the SHM?", default_to_yes=True
```

Comment:
```suggestion
            "Do you wish to continue tearing down the SHM?", default_to_yes=False
```
I think it would make sense to default to no when we are adding a protection like this.